deepchem/models/torch_models/tests/test_gan.py

```diff
@@ -169,3 +204,123 @@ def test_get_noise_batch():
                      create_discriminator(data_shape, conditional_shape))
     noise = gan.get_noise_batch(batch_size)
     assert noise.shape == (gan.noise_input_shape)
+
+
+@pytest.mark.torch
+def generate_batch(batch_size):
+    """Draw training data from a Gaussian distribution, where the mean  is a conditional input."""
+    means = 10 * np.random.random([batch_size, 1])
+    values = np.random.normal(means, scale=2.0)
+    return means, values
+
+
+@pytest.mark.torch
+def generate_data(gan, batches, batch_size):
+    for _ in range(batches):
+        means, values = generate_batch(batch_size)
+        batch = {gan.data_inputs[0]: values, gan.conditional_inputs[0]: means}
+        yield batch
```

Comment:
This function is where the dictionary is being created

Reply:
Ok, so you are mapping the torch tensor to a numpy value?

Reply:
yup

Reply:
Two questions:

- How does the TensorFlow implementation handle it?
- Can we just give inputs a unique name and bypass the issue? Say input1, input2, etc are the names. Then we can just use input name

Reply:
1. The TF implementation uses `tensor.ref()` to generate a reference to the tensor and there is no pytorch equivalent for the same, 
2. we could give it a unique name, but there could be issues with which layer created that specific tensor, as that is the key differentiating factor in the TF model

Reply:
Proceed with a prototype implementation of 2. It's OK to change the user API a bit here if needed (having the user explicitly name inputs for example). We will have a deprecation cycle